hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoader.java

```diff
@@ -439,4 +451,14 @@ public String toString() {
     public void onKeyLoad(ExecutionCallback<Boolean> callback) {
         loadFinished.andThen(callback, execService.getExecutor(MAP_LOAD_ALL_KEYS_EXECUTOR));
     }
+
+    public void promoteToLoadedOnMigration() {
+        state.next(State.LOADING);
+        state.next(State.LOADED);
```

Comment:
Ok. Maybe at least we can add your [statement](https://github.com/hazelcast/hazelcast/pull/7992#discussion_r61106721) to the `promoteToLoadedOnMigration ` method for the future reviewers? Because at first glance `state.next(State.LOADED)` seems enough.